packages/flutter_tools/lib/src/web/bootstrap.dart

```diff
@@ -4,38 +4,19 @@
 
 import 'package:package_config/package_config.dart';
 
-const String _currentDirectoryScript = r"""
-var _currentDirectory = (function () {
-  var lines = new Error().stack.split('\n');
-  function lookupUrl() {
-    if (lines.length > 2) {
-      var match = lines[1].match(/^\s+at (.+):\d+:\d+.*$/);
-      // Chrome.
-      if (match) return match[1];
-      // Chrome nested eval case.
-      match = lines[1].match(/^\s+at eval [(](.+):\d+:\d+[)]$/);
-      if (match) return match[1];
-      // Edge.
-      match = lines[1].match(/^\s+at.+\((.+):\d+:\d+\)$/);
-      if (match) return match[1];
-      // Firefox.
-      match = lines[0].match(/[<][@](.+):\d+:\d+$/);
-      if (match) return match[1];
-    }
-    // Safari.
-    return lines[0].match(/[@](.+):\d+:\d+$/)[1];
-  }
-  var _url = lookupUrl();
-  var lastSlash = _url.lastIndexOf('/');
-  if (lastSlash == -1) return _url;
-  var currentDirectory = _url.substring(0, lastSlash + 1);
-  return currentDirectory;
-})();
-""";
+String generateDDCBootstrapScript({
+  required String entrypoint,
+  required String ddcModuleLoaderUrl,
+  required String mapperUrl,
+  required bool generateLoadingIndicator,
+}) {
+  return '''
+${generateLoadingIndicator ? _generateLoadingIndicator() : ""}
+// TODO(markzipan): This is safe if Flutter app roots are always equal to the
+// host root. Validate if this is true.
+var _currentDirectory = '/';
```

Comment:
Can we move the value of `_currentDirectory` as an optional, named parameter of `generateDDCBootstrapScript` that defaults to `'/'`? If it can come from the build system, it should! :P

Note that for `flutter build`, something similar to this parameter already exists: it's [`base-href`](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/commands/build_web.dart#L42); I guess if needed, it could be added for `flutter run` as well?

Reply:
Good call - I can add that! I'll revisit this if the root needs to change later on, but I suspect this'll suffice for now.